kubernetes/kubernetes.go

```diff
@@ -186,14 +185,18 @@ func (ka KubeAgentConfig) collectMetrics(
 	// get raw Heapster metric sample
 	hme, err := rawClient.GetRawEndPoint("heapster-metrics-export", metricSampleDir, config.HeapsterURL)
 	if err != nil {
-		return fmt.Errorf("error retrieving Raw Heapster metrics: %s", err)
+		return fmt.Errorf("unable to retrieve raw heapster metrics: %s", err)
 	}
 
 	defer util.SafeClose(hme.Close, &rerr)
 
-	err = handleBaselineHeapsterMetrics(msd, baselineMetricSample, hme.Name())
-	if err != nil {
-		return err
+	baselineMetricSample, fileCount, err := util.MatchOneFile(
+		path.Dir(config.msExportDirectory.Name()), "/baseline-metrics-export*")
+	if err == nil || fileCount == 0 {
```

Comment:
so if err != nil but fileCount is zero we still want to continue?

Reply:
yes. per your suggestion I refactored for better read-ability